### PR TITLE
Refactored streaming to remove the confusing double "conn"

### PR DESCRIPTION
**Pull request by AsafMah**

### CODE REVIEW

File: kusto/conn.go [hunk 1]
```diff
@@ -183,39 +169,81 @@ func (c *conn) doRequest(ctx context.Context, execType int, db string, query Stm
 		return 0, nil, nil, nil, errors.ES(op, errors.KInternal, "internal error: did not understand the type of execType: %d", execType)
 	}
 
+	headers := c.getHeaders(properties)
+	responseHeaders, closer, err := c.doRequest(ctx, op, endpoint, io.NopCloser(buff), headers, fmt.Sprintf("With query: %s", query.String()))
+	return op, headers, responseHeaders, closer, err
+}
+
+func (c *conn) doRequest(
```
Comment (yogilad) [inline kusto/conn.go:182]: xxxImpl

File: kusto/conn.go [hunk 2]
```diff
@@ -132,7 +132,7 @@ func (c *conn) execute(ctx context.Context, execType int, db string, query Stmt,
 	return execResp{reqHeader: reqHeader, respHeader: respHeader, frameCh: frameCh}, nil
 }
 
-func (c *conn) doRequest(ctx context.Context, execType int, db string, query Stmt, properties requestProperties) (errors.Op, http.Header, http.Header,
+func (c *conn) doRequestFromType(ctx context.Context, execType int, db string, query Stmt, properties requestProperties) (errors.Op, http.Header, http.Header,
```
Comment (yogilad) [inline kusto/conn.go:138]: Keep it and change the API downstream to xxxImpl

File: kusto/ingest/ingest.go [hunk 3]
```diff
@@ -225,15 +227,15 @@ func (i *Ingestion) Stream(ctx context.Context, payload []byte, format DataForma
...
+func (i *Ingestion) getStreamConn() (streamIngestor, error) {
 	i.connMu.Lock()
 	defer i.connMu.Unlock()
 
 	if i.streamConn != nil {
 		return i.streamConn, nil
 	}
 
-	sc, err := conn.New(i.client.Endpoint(), i.client.Auth(), i.client.HttpClient(), i.client.ClientDetails())
+	sc, err := kusto.NewConn(strings.Replace(i.client.Endpoint(), "ingest-", "", 1), i.client.Auth(), i.client.HttpClient(), i.client.ClientDetails())
```
Comment (yogilad) [inline kusto/ingest/ingest.go:238]: Another PR leaking in or is it intentional?

File: kusto/conn_streaming_ingest.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	"io"
+	"net/url"
+)
+
+type DataFormatForStreaming interface {
+	CamelCase() string
+	ForceDefault() DataFormatForStreaming
+}
+
+func (c *conn) StreamIngest(ctx context.Context, db, table string, payload io.Reader, format DataFormatForStreaming, mappingName string, clientRequestId string) error {
```
Comment (yogilad) [inline kusto/conn_streaming_ingest.go:16]: Stream?

File: kusto/ingest/streaming.go [hunk 5]
```diff
@@ -15,7 +17,7 @@ import (
 
 type streamIngestor interface {
```
Comment (yogilad) [inline kusto/ingest/streaming.go:18]: streamConn

**Review by yogilad — APPROVED** [review]

File: kusto/conn_streaming_ingest.go (hunk 4 above)
Comment (AsafMah) [reply kusto/conn_streaming_ingest.go:16]: Why? it's doing streaming ingestion. calling it stream will make you think it's a streaming query

File: kusto/ingest/streaming.go (hunk 5 above)
Comment (AsafMah) [reply kusto/ingest/streaming.go:18]: Why? it's a good name for what the interface does. I hate the "conn" terminology.

File: kusto/ingest/ingest.go (hunk 3 above)
Comment (AsafMah) [reply kusto/ingest/ingest.go]: I moved it to a function

File: kusto/conn_streaming_ingest.go (hunk 4 above)
Comment (yogilad) [inline kusto/conn_streaming_ingest.go:16]: CID is not added to the headers

**Pull request merged**